manticore/native/memory.py

```diff
@@ -892,6 +892,19 @@ def perms(self, index):
         else:
             return self.map_containing(index).perms
 
+    def max_exec_size(self, index, size):
```

Comment:
This needs a docstring and I we should probably change the argument names to `addr` and maybe `offset`?

My reasoning is that we pass here _some address_ and we ask _what is the maximum offset from that address that is executable_. This being said maybe the function name should be `max_exec_offset`? I am not sure here.

Reply:
Btw this function might be handy for optimizing other memory accesses - read/writes - so maybe it would be best to pass the permission flag here too?

Reply:
> maybe offset

It is a size, not an offset

> Btw this function might be handy for optimizing other memory accesses

Which ones ?
Should this be part of this PR ?

Reply:
> Which ones ?

At first I thought about all usages of `memory.access_ok` but it turns out it accepts a `slice` too!

Anyway, this: https://github.com/trailofbits/manticore/blob/master/manticore/native/memory.py#L1201-L1202 could probably be fixed/optimized out to do the check on a slice instead of single bytes.

At least maybe, because that is symbolic and can lead to some other issues? I am not sure here.

> Should this be part of this PR ?

Not really. I was thinking that we could change the function to accept the permission flag as argument but since `accept_ok` accepts a `slice` and is used like this in some places already, then maybe not.